src/components/input/test/SelectNext-test.js

```diff
@@ -0,0 +1,220 @@
+        .exists('[data-testid="selected-option"]');
+
+    assert.isFalse(isOptionSelected(1));
+    assert.isFalse(isOptionSelected(2));
+    assert.isTrue(isOptionSelected(3));
+    assert.isFalse(isOptionSelected(4));
+    assert.isFalse(isOptionSelected(5));
+  });
+
+  it('marks the right item as disabled', () => {
+    const wrapper = createComponent();
+    const isOptionDisabled = id =>
+      wrapper
+        .find(`[data-testid="option-${id}"]`)
+        .exists('[data-testid="disabled-option"]');
+
+    assert.isFalse(isOptionDisabled(1));
+    assert.isFalse(isOptionDisabled(2));
+    assert.isFalse(isOptionDisabled(3));
+    assert.isTrue(isOptionDisabled(4));
+    assert.isFalse(isOptionDisabled(5));
+  });
+
+  it('toggles listbox when button is clicked', () => {
+    const wrapper = createComponent();
+
+    assert.isTrue(isListboxClosed(wrapper));
+    toggleListbox(wrapper);
+    assert.isFalse(isListboxClosed(wrapper));
+  });
+
+  it('closes listbox when Escape is pressed', () => {
+    const wrapper = createComponent();
+
+    toggleListbox(wrapper);
+    assert.isFalse(isListboxClosed(wrapper));
+
+    document.body.dispatchEvent(
+      new KeyboardEvent('keydown', { key: 'Escape' }),
+    );
+    wrapper.update();
+
+    // Listbox is closed after `Escape` is pressed
+    assert.isTrue(isListboxClosed(wrapper));
+  });
+
+  it('closes listbox when clicking away', () => {
+    const wrapper = createComponent();
+
+    toggleListbox(wrapper);
+    assert.isFalse(isListboxClosed(wrapper));
+
+    const externalButton = document.createElement('button');
+    document.body.append(externalButton);
+
+    externalButton.click();
+    wrapper.update();
+
+    try {
+      // Listbox is closed after other element is clicked
+      assert.isTrue(isListboxClosed(wrapper));
+    } finally {
+      externalButton.remove();
+    }
+  });
+
+  it('restores focus to toggle button after closing listbox', () => {
+    const wrapper = createComponent();
+    toggleListbox(wrapper);
+
+    // Focus body before closing listbox
+    document.body.focus();
+    toggleListbox(wrapper);
+    wrapper.update();
+
+    assert.equal(document.activeElement, getToggleButton(wrapper).getDOMNode());
+  });
+
+  it('displays listbox when ArrowDown is pressed on toggle', () => {
+    const wrapper = createComponent();
+
+    assert.isTrue(isListboxClosed(wrapper));
+
+    getToggleButton(wrapper)
+      .getDOMNode()
+      .dispatchEvent(new KeyboardEvent('keydown', { key: 'ArrowDown' }));
+    wrapper.update();
+
+    assert.isFalse(isListboxClosed(wrapper));
+  });
+
+  [
+    { windowHeight: 10000, containerPaddingTop: 0, shouldDropUp: false },
+    { windowHeight: 200, containerPaddingTop: 1000, shouldDropUp: true },
+  ].forEach(({ windowHeight, containerPaddingTop, shouldDropUp }) => {
+    it('makes listbox drop up or down based on available space below', () => {
+      window.innerHeight = windowHeight;
```

Comment:
Huh, I'm slightly surprised that setting `window.innerHeight` works. This could potentially impact other tests, so we might want to reset it afterwards, or add a test seam.

Reply:
Actually, I forgot I wanted to reset it to its previous value afterwards, otherwise it will keep the modified value for all tests coming after.

I'm not sure if a test seam would work here, because the bounding client rects from the other elements are affected by the window size and where are they rendered inside of it, but I will try just in case.

Reply:
I realized adding padding to the container is enough. There's no need to change the window height during the test.